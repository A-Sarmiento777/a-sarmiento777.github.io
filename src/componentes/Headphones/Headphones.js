import React, { useEffect, useState } from "react";
//import { apiURL } from "../../Utils/ApiUrl";
import Headphone from "../Headphone/Headphone";

const Headphones = ({ agregarProductoAlCarrito, agregarProductoAlCarritoFav, removeItemFromCartFav, cart, favProducts }) => {
  const [headphones, setHeadphones] = useState([]);

  async function getHeadphones() {
    try {
      //let response = await fetch(apiURL + "api/headphones");
      let response = await fetch("/api/headphones");
      let data = await response.json();
      if (data) {
        setHeadphones(data);
      }
    } catch (error) {
      console.log(error);
    }
  }

  useEffect(() => {
    getHeadphones();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  return (
    <div className="pt-4 text-center" data-testid="headphones1">
      <h1>Headphones</h1>
      <div className="d-flex flex-wrap justify-content-center">
        {headphones &&
          headphones.map((headphone, index) => {
            return (
				<Headphone agregarProductoAlCarrito={agregarProductoAlCarrito} agregarProductoAlCarritoFav={agregarProductoAlCarritoFav} removeItemFromCartFav={removeItemFromCartFav} cart={cart} favProducts={favProducts} headphone={headphone}/>
            );
          })}
      </div>
    </div>
  );
};

export default Headphones;
