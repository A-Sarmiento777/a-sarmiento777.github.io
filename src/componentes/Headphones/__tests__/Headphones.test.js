import React from "react";
import { act, create, waitFor } from "react-test-renderer";
import Headphones from "../Headphones";

test("It should render completed Headphones", async () => {
  jest.spyOn(global, "fetch").mockImplementation(() => {
    return Promise.resolve({
      json: () => Promise.resolve([]),
    });
  });

  let testRenderer;
  await act(async () => {
    testRenderer = create(
      <Headphones agregarProductoAlCarrito={() => {}} agregarProductoAlCarritoFav={() => {}} removeItemFromCartFav={() => {}} cart={[]} favProducts={[]} />
    );
  });

  let tree = testRenderer.toJSON();
  expect(tree).toMatchSnapshot();
  expect(global.fetch).toHaveBeenCalledTimes(1);
});
