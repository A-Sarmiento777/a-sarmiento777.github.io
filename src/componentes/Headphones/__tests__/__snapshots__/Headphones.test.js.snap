// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render completed Headphones 1`] = `
<div
  className="pt-4 text-center"
  data-testid="headphones1"
>
  <h1>
    HÃ¶rlurar
  </h1>
  <div
    className="d-flex flex-wrap justify-content-center"
  />
</div>
`;
